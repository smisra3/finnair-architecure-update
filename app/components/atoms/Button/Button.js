// @flow
import type { Node } from 'react';

import styled from 'styled-components';
import styles from './Button.style';
import Spinner from '../../atoms/Spinner';

type Props = {
  className: string,
  children: Node,
  inheritedStyles?: string,
  type?: string,
  primary?: boolean,
  secondary?: boolean,
  elementRef?: any => void,
  loader?: boolean,
  loaderPositionCenter?: boolean,
  disabled?: boolean,
  ariaLabel?: string,
};

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  elementRef,
  loader,
  loaderPositionCenter,
  disabled,
  ariaLabel,
  ...others
}: Props): Node => (
  <button
    className={className}
    type={type}
    {...others}
    ref={elementRef}
    disabled={disabled || loader}
    aria-label={ariaLabel || null}
  >
    {loader && (
    <div className={`spinner ${loaderPositionCenter ? 'center-spinner' : ''}`}>
      <Spinner overlayEnabled spinner />
    </div>
      )}
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  elementRef: undefined,
  loader: false,
  disabled: false,
  loaderPositionCenter: false,
  ariaLabel: '',
};

export default styled(Button)`${styles}`;
export { Button as ButtonVanilla };
