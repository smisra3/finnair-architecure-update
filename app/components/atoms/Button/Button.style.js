import { css } from 'styled-components';

const ButtonStyles = css`
  border: none;
  border-radius: 0;
  background: none;
  position: relative;
  min-height: 48px;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? '0.3' : '1')};
  ${props =>
    (props.primary
      ? `
        background: ${props.theme.buttonBackgroundColorPrimary} ;
        color: ${props.theme.buttonColorPrimary};
        font-size: ${props.theme.fontSizeButtonPrimary};
        padding: 0 1rem;
      ` : '')};
  ${props =>
    (props.secondary
      ? `
        background: ${props.theme.buttonBackgroundColorSecondary} ;
        color: ${props.theme.buttonColorSecondary};
        font-size: ${props.theme.fontSizeButtonSecondary};
        padding: 0 1rem;
      ` : '')};
  ${props =>
    (props.fullWidth
      ? `
        width: 100% ;
      ` : '')};

  &:focus {
    outline: 1px dashed ${props => props.theme.colorOutline};
  }

  &:focus:not(.focus-visible) {
    outline: none;
  }

  .spinner {
    position: absolute;
    top: 50%;
    left: 9px;
    transform: translateY(-50%);

    &.center-spinner {
      left: 50%;
      transform: translateX(-50%) translateY(-50%);
    }

    div {
      height: 25px;
      width: 25px;
      top: 0;
    }
  }
`;

export default ButtonStyles;
