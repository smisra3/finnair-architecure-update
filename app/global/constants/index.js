/* TODO: Remove lint exception */
/* eslint-disable import/prefer-default-export */
export const DEVICE_TYPE = 'DEVICE_TYPE';
export const IS_TABLET = 'IS_TABLET';
export const REGION_TYPE = 'REGION_TYPE';
export const CONFIG_KEYS = 'CONFIG_KEYS';
export const CURRENT_ROUTE = 'CURRENT_ROUTE';
export const PAGE_URL = 'PAGE_URL';
export const PAGE_QUERY = 'PAGE_QUERY';
export const GLOBAL_DATA_FAILURE = 'FIL/GLOBAL/GLOBAL_DATA_FAILURE';
export const CHECK_AUTH = 'FIL/GLOBAL/CHECK_AUTH';
export const CHECK_AUTH_SUCCESS = 'FIL/GLOBAL/CHECK_AUTH_SUCCESS';
export const SET_USER_STATE = 'FIL/GLOBAL/SET_USER_STATE';
export const PAGE_ORIGIN = 'PAGE_ORIGIN';
export const LABELS_STATE_KEY = 'labels';

export const USER_STATE_LOGGED_IN = 'FIL/GLOBAL/SUCCESS';
export const USER_STATE_RECOGNIZED = 'FIL/GLOBAL/RECOGNIZED';
export const USER_STATE_ANONYMOUS = 'FIL/GLOBAL/ANONYMOUS';

export const GET_LABELS = 'FIL/GLOBAL/GET_LABELS';
export const GET_LABELS_SUCCESS = 'FIL/GLOBAL/GET_LABELS_SUCCESS';
export const GET_LABELS_FAILURE = 'FIL/GLOBAL/GET_LABELS_FAILURE';
