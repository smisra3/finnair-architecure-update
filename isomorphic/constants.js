const HEARTBEAT_SESSION_NAME = 'JSESSIONID';
const ENV_DEVELOPMENT = 'development';
const ENV_PRODUCTION = 'production';
const SERVICE_WORKER_FILE = '/service-worker.js';
const API_PROXY_PATH = '/api';
const CLEAR_SERVICES_CACHE = '/clear-services-cache';
const HEALTH_CHECK = '/healthcheck';
const PROTOCOL_HTTPS = 'https';
const RESPONSE_OK = 200;
const RESPONSE_INTERNAL_SERVER_ERROR = 500;
const STATIC_FOLDER_PREFIX = '/static/';

module.exports = {
  HEARTBEAT_SESSION_NAME,
  ENV_DEVELOPMENT,
  ENV_PRODUCTION,
  SERVICE_WORKER_FILE,
  API_PROXY_PATH,
  CLEAR_SERVICES_CACHE,
  PROTOCOL_HTTPS,
  RESPONSE_OK,
  RESPONSE_INTERNAL_SERVER_ERROR,
  STATIC_FOLDER_PREFIX,
  HEALTH_CHECK,
};
