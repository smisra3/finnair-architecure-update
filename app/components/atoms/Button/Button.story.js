// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  primaryButton,
  primaryButtonDisabled,
  secondaryButton,
  secondaryButtonDisabled,
} from './Button.mock';

// Import Vanilla Component to display markup and props
import { ButtonVanilla } from './Button';
// Import Styled Component to showcase variations
import Button from './index';

storiesOf('Atoms', module).addWithChapters('Buttons', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <ButtonVanilla {...primaryButton} className={`hide-default-sample ${primaryButton.className}`}>Submit</ButtonVanilla>,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary Button Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Button {...primaryButton}>Submit</Button>,
        },
        {
          title: 'Disabled',
          sectionFn: () => <Button {...primaryButtonDisabled}>Submit</Button>,
        },
      ],
    },
    {
      title: 'Secondary Button Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Button {...secondaryButton}>Submit</Button>,
        },
        {
          title: 'Disabled',
          sectionFn: () => <Button {...secondaryButtonDisabled}>Submit</Button>,
        },
      ],
    },
  ],
});
