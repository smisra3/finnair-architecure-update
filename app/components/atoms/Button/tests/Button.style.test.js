import { shallow } from 'enzyme';
import 'jest-styled-components';

import Button from '../';
import Theme from '../../../../styles/theme';
import { DESKTOP, MOBILE } from '../../../../constants';

describe('Button', () => {
  let button = '';
  const theme = { ...Theme };

  test('should render correctly', () => {
    theme.deviceType = DESKTOP;
    button = shallow(<Button theme={theme} />);
    expect(button).toMatchSnapshot();
  });

  test('should render correctly for mobile', () => {
    theme.deviceType = MOBILE;
    button = shallow(<Button theme={theme} />);
    expect(button).toMatchSnapshot();
  });
});
