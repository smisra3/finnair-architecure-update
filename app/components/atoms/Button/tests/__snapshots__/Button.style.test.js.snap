// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
.c0 {
  border: none;
  border-radius: 0;
  background: none;
  position: relative;
  min-height: 48px;
  cursor: pointer;
  opacity: 1;
}

.c0:focus {
  outline: 1px dashed #999;
}

.c0:focus:not(.focus-visible) {
  outline: none;
}

.c0 .spinner {
  position: absolute;
  top: 50%;
  left: 9px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 .spinner.center-spinner {
  left: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.c0 .spinner div {
  height: 25px;
  width: 25px;
  top: 0;
}

<Button
  ariaLabel=""
  className="c0"
  disabled={false}
  inheritedStyles=""
  loader={false}
  loaderPositionCenter={false}
  primary={false}
  secondary={false}
  theme={
    Object {
      "backgroundColorDisabled": "grey",
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "cautionBackgroundColor": "#fffdf3",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "deviceType": "desktop",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "letterSpacing": "1px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zFilter": 100,
        "zFilterActions": 400,
        "zFilterList": 150,
        "zFlyout": 200,
        "zLoader": 1000,
        "zMenu": 800,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zPlacesAutocomplete": 1100,
        "zSticky": 700,
      },
    }
  }
  type="button"
/>
`;

exports[`Button should render correctly for mobile 1`] = `
.c0 {
  border: none;
  border-radius: 0;
  background: none;
  position: relative;
  min-height: 48px;
  cursor: pointer;
  opacity: 1;
}

.c0:focus {
  outline: 1px dashed #999;
}

.c0:focus:not(.focus-visible) {
  outline: none;
}

.c0 .spinner {
  position: absolute;
  top: 50%;
  left: 9px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 .spinner.center-spinner {
  left: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.c0 .spinner div {
  height: 25px;
  width: 25px;
  top: 0;
}

<Button
  ariaLabel=""
  className="c0"
  disabled={false}
  inheritedStyles=""
  loader={false}
  loaderPositionCenter={false}
  primary={false}
  secondary={false}
  theme={
    Object {
      "backgroundColorDisabled": "grey",
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "cautionBackgroundColor": "#fffdf3",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "deviceType": "mobile",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "letterSpacing": "1px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zFilter": 100,
        "zFilterActions": 400,
        "zFilterList": 150,
        "zFlyout": 200,
        "zLoader": 1000,
        "zMenu": 800,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zPlacesAutocomplete": 1100,
        "zSticky": 700,
      },
    }
  }
  type="button"
/>
`;
